File: Design Patterns/Behavioral/Strategy/Case 1/Python/MallardDuck.py
from Duck import Duck
from Fly.FlyBehavior import FlyBehavior
from Fly.FlyWithWings import FlyWithWings
from Quak.QuackBehavior import QuackBehavior
from Quak.Quak import Quak

class MallardDuck(Duck,FlyBehavior,QuackBehavior):

    def __init__(self) -> None:
        Duck.__init__(self)
        self.setFlyBehavior(FlyWithWings())
        self.setQuakBehavior(Quak())
    
    def fly(self):
        self.getFlyBehavior().fly()

    def quak(self):
        self.getQuakBehavior().quak()

    def dispay():
        pass

    def swim():
        pass

File: Design Patterns/Behavioral/Strategy/Case 1/PHP/MallardDuck.php
<?php
namespace PHP;
require_once("Quak/QuakBehavior.php");
use PHP\Quak\QuakBehavior;
require_once("Fly/FlyBehavior.php");
use PHP\Fly\FlyBehavior;
require_once("Fly/FlyWithWings.php");
use PHP\Fly\FlyWithWings;
require_once("Quak/Quak.php");
use PHP\Quak\Quak;
require_once("Duck.php");
use PHP\Duck;


class MallardDuck extends Duck implements QuakBehavior,FlyBehavior{
    
    function __construct()
    {
        $this->setFlyBehavior(new FlyWithWings());
        $this->setQuakBehavior(new Quak());    
    }

    function fly(){
        $this->getFlyBehavior()->fly();
    }

    function quak(){
        $this->getQuakBehavior()->quak();
    }

    function display(){
        echo 'displaying MallardDuck';
    }
}